src/routes/secret_test.go

```diff
@@ -0,0 +1,139 @@
+package routes_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/dana-team/platform-backend/src/routes"
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/gin-gonic/gin"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+var (
+	router *gin.Engine
+	client *fake.Clientset
+)
+
+var _ = BeforeSuite(func() {
+	client = fake.NewSimpleClientset()
+	logger, _ := zap.NewProduction()
+	router = setupRouter(client, logger)
+
+	createTestSecret()
+})
+
+var _ = Describe("Secrets Routes", func() {
+	It("should create a secret successfully", func() {
```

Comment:
We are not testing for failures here, only for success

Reply:
Alright, I'll add test cases to cover failure scenarios.

Reply:
It requires more work, and probably using proper gin's validation as described in #13. Let's take care of it later.